src/main/java/org/truffleruby/parser/lexer/RubyLexer.java

```diff
@@ -3493,12 +3493,12 @@ public boolean update_heredoc_indent(int c) {
     }
 
     public void validateFormalIdentifier(Rope identifier) {
-        int first = identifier.get(0) & 0xFF;
-
-        if (Character.isUpperCase(first)) {
+        if (isFirstCharacterEncodingAwareUppercase(identifier)) {
             compile_error("formal argument cannot be a constant");
         }
 
+        int first = identifier.getString().charAt(0);
```

Comment:
We should still use `int first = identifier.get(0) & 0xFF;` here to avoid creating an intermediate Java String.